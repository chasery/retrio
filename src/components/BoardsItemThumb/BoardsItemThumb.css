.BoardsItemThumb {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  column-gap: 0.5rem;
  grid-template-rows: 1fr;
  padding: 1rem;
}

.BoardsItemThumb__card {
  width: 100%;
  margin-bottom: 0.25rem;
  padding-top: 56.25%;
  align-self: stretch;
}
.BoardsItemThumb__card.filled {
  background-color: #eef4fb;
}
.BoardsItemThumb__card.empty {
  background-color: transparent;
}
