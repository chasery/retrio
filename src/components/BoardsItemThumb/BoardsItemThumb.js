import React from 'react';
import './BoardsItemThumb.css';

function BoardsItemThumb(props) {
  const createCards = () => {
    const random = Math.floor(Math.random() * 5);
    const cards = [];

    for (let i = 0; i < 4; i++) {
      if (i < random) {
        cards.push(
          <div key={i} className='BoardsItemThumb__card filled'></div>
        );
      } else {
        cards.push(<div key={i} className='BoardsItemThumb__card empty'></div>);
      }
    }

    return cards;
  };

  return (
    <div className='BoardsItemThumb'>
      <div className='BoardsItemThumb__column'>{createCards()}</div>
      <div className='BoardsItemThumb__column'>{createCards()}</div>
      <div className='BoardsItemThumb__column'>{createCards()}</div>
      <div className='BoardsItemThumb__column'>{createCards()}</div>
    </div>
  );
}

export default BoardsItemThumb;
