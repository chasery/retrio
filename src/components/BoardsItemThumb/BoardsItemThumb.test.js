import React from 'react';
import { shallow } from 'enzyme';
import BoardsItemThumb from './BoardsItemThumb';

describe('BoardsItemThumb component', () => {
  it('renders without error', () => {
    shallow(<BoardsItemThumb />);
  });
});
