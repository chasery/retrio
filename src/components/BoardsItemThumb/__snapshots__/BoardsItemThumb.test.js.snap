// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardsItemThumb component renders without error 1`] = `
<div
  className="BoardsItemThumb"
>
  <div
    className="BoardsItemThumb__column"
  >
    <div
      className="BoardsItemThumb__card empty"
      key="0"
    />
    <div
      className="BoardsItemThumb__card empty"
      key="1"
    />
    <div
      className="BoardsItemThumb__card empty"
      key="2"
    />
    <div
      className="BoardsItemThumb__card empty"
      key="3"
    />
  </div>
  <div
    className="BoardsItemThumb__column"
  >
    <div
      className="BoardsItemThumb__card filled"
      key="0"
    />
    <div
      className="BoardsItemThumb__card empty"
      key="1"
    />
    <div
      className="BoardsItemThumb__card empty"
      key="2"
    />
    <div
      className="BoardsItemThumb__card empty"
      key="3"
    />
  </div>
  <div
    className="BoardsItemThumb__column"
  >
    <div
      className="BoardsItemThumb__card filled"
      key="0"
    />
    <div
      className="BoardsItemThumb__card filled"
      key="1"
    />
    <div
      className="BoardsItemThumb__card empty"
      key="2"
    />
    <div
      className="BoardsItemThumb__card empty"
      key="3"
    />
  </div>
  <div
    className="BoardsItemThumb__column"
  >
    <div
      className="BoardsItemThumb__card filled"
      key="0"
    />
    <div
      className="BoardsItemThumb__card filled"
      key="1"
    />
    <div
      className="BoardsItemThumb__card filled"
      key="2"
    />
    <div
      className="BoardsItemThumb__card empty"
      key="3"
    />
  </div>
</div>
`;
